parser_library/src/semantics/statement.h

```diff
@@ -158,6 +158,37 @@ struct statement_si_defer_done : public complete_statement
     const range& stmt_range_ref() const override { return deferred_stmt->stmt_range_ref(); }
 };
 
+struct preprocessor_statement_si : public statement_si
```

Comment:
This is just a first read reaction from me, but I am wondering whether it is really needed to model this preprocessor output using the `*_si` object with all their baggage...

Reply:
I was thinking something similar. This approach does seem to heavy for Endevor statements but when dealing with CICS it seemed more appropriate. I'll add a todo note to review this part when dealing with DB2 statements and possibly change it